deps/jerry/jerry-core/ecma/operations/ecma-exceptions.c

```diff
@@ -134,6 +135,32 @@ ecma_new_standard_error (ecma_standard_error_t error_type) /**< native error typ
 
   ((ecma_extended_object_t *) new_error_obj_p)->u.class_prop.class_id = LIT_MAGIC_STRING_ERROR_UL;
 
+  const char *frames_id_p = "__frames__";
+
+  ecma_string_t *frames_str_p = ecma_new_ecma_string_from_utf8 ((const lit_utf8_byte_t *) frames_id_p, 10);
+
+  ecma_property_value_t *prop_value_p = ecma_create_named_data_property (new_error_obj_p,
+                                                                         frames_str_p,
+                                                                         ECMA_PROPERTY_CONFIGURABLE_WRITABLE,
+                                                                         NULL);
+  ecma_deref_ecma_string (frames_str_p);
+
+  uint32_t depth = 10;
+  // create frames
+  uint32_t* frames = malloc(sizeof(uint32_t) * depth);
+  memset(frames, 0, sizeof(uint32_t) * depth);
+  jerry_get_stacktrace_depth(frames, depth);
+
+  jerry_value_t jframes = jerry_create_array(depth);
+  for (uint32_t i = 0; i < depth; ++i) {
+    jerry_set_property_by_index(jframes, i, jerry_create_number(frames[i]));
+  }
+
+  free(frames);
+
+  prop_value_p->value = jframes;
+  ecma_deref_object (ecma_get_object_from_value (jframes));
+
```

Comment:
Since error stack trace has to be started with `JERRY_CONTEXT.vm_top_context_p`, a `jerry_*` api has to be called in this case.
Or is there any way can pass through `JERRY_CONTEXT` to access current top vm context?

Reply:
See the above comments of mine, I think in ecma, call jcontext looks acceptable. Then we need to move some code into [jcontext.h](https://github.com/Rokid/ShadowNode/blob/e6ced3744bf3bcc8d44ef90edaafebb545b5012c/deps/jerry/jerry-core/jcontext/jcontext.h), and call those from there :)